src/hotspot/share/opto/phaseX.cpp

```diff
@@ -85,15 +110,12 @@ Node *NodeHash::hash_find( const Node *n ) {
   while( 1 ) {                  // While probing hash table
     if( k->req() == req &&      // Same count of inputs
         k->Opcode() == op ) {   // Same Opcode
-      for( uint i=0; i<req; i++ )
-        if( n->in(i)!=k->in(i)) // Different inputs?
-          goto collision;       // "goto" is a speed hack...
-      if( n->cmp(*k) ) {        // Check for any special bits
+      bool collision = check_for_collision(n, k);
+      if (collision == false && n->cmp(*k)) {  // Check for any special bits
         NOT_PRODUCT( _lookup_hits++ );
         return k;               // Hit!
       }
     }
-  collision:
```

Comment:
I would replace this line with a comment like this:
`// k was not a hit. Find another candidate and try again.`